controller/engine_controller.go

```diff
@@ -907,15 +924,18 @@ func (m *EngineMonitor) refresh(engine *longhorn.Engine) error {
 		return err
 	}
 	if needClone {
-		if err = cloneSnapshot(engine, client, m.ds); err != nil {
+		if err = cloneSnapshot(engine, engineClientProxy, m.ds); err != nil {
 			return err
 		}
 	}
 
 	return nil
 }
 
-func preRestoreCheckAndSync(log logrus.FieldLogger, engine *longhorn.Engine, rsMap map[string]*longhorn.RestoreStatus, addressReplicaMap map[string]string, cliAPIVersion int, client engineapi.EngineClient, ds *datastore.DataStore) (needRestore bool, err error) {
+func preRestoreCheckAndSync(log logrus.FieldLogger, engine *longhorn.Engine,
```

Comment:
Does this function still need `client *engineapi.EngineBinary`? The caller will input the compatible client (returned from `proxyHandler.GetClient(im, engineCliClient)`), which may be a binary one.
Similar below.

Reply:
Removed.